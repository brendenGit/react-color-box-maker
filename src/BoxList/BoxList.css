.BoxList {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: auto;
    width: 80%;
    flex-wrap: wrap;
}