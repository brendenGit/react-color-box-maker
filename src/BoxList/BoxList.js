import { useState } from "react";
import { v4 as uuid } from 'uuid';

import Box from "../Box/Box";
import NewBoxForm from "../NewBoxForm/NewBoxForm";

import "./BoxList.css";

function BoxList() {
    const [boxes, setBoxList] = useState([]);

    const addBox = ({ color, width, height }) => {
        const newBox = { color, width, height, id: uuid() }
        setBoxList(oldBoxes => ([...oldBoxes, newBox]));
    };
 
    const removeBox = (id) => {
        setBoxList(oldBoxes => (oldBoxes.filter(box => box.id !== id)));
    };

    return (
        <div>
            <NewBoxForm addBox={addBox} />
            <div className="BoxList" key={uuid()}>
                {boxes.map(box => <Box 
                                    color={box.color} 
                                    width={box.width} 
                                    height={box.height} 
                                    id={box.id}
                                    key={uuid()}
                                    removeBox={removeBox}
                                />)}
            </div>
        </div>
    )
};

export default BoxList;