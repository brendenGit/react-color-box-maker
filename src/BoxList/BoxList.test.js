import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

//Smoke test
it("renders without crashing", function () {
    render(<BoxList />)
});

//Snapshot test
it("matches snapshot", function () {
    const { asFragment } = render(<BoxList />)
    expect(asFragment()).toMatchSnapshot();
});

test('adds and removes boxes', () => {
    const { container, getByLabelText, getByText } = render(<BoxList />);

    // Add a box
    fireEvent.change(getByLabelText('Color:'), { target: { value: 'red' } });
    fireEvent.change(getByLabelText('Width:'), { target: { value: '200' } });
    fireEvent.change(getByLabelText('Height:'), { target: { value: '200' } });
    fireEvent.click(getByText('Submit'));

    // Check if the box is added
    const addedBox = container.querySelector('.Box');
    expect(addedBox).toBeInTheDocument();

    // Remove the added box
    fireEvent.click(addedBox);

    // Check if the box is removed
    expect(addedBox).not.toBeInTheDocument();
});