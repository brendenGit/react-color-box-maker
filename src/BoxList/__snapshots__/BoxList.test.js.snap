// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="color"
      >
        Color:
        <input
          id="color"
          name="color"
          type="text"
          value=""
        />
      </label>
      <br />
      <label
        for="width"
      >
        Width:
        <input
          id="width"
          name="width"
          type="number"
          value=""
        />
      </label>
      <br />
      <label
        for="height"
      >
        Height:
        <input
          id="height"
          name="height"
          type="number"
          value=""
        />
      </label>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
    <div
      class="BoxList"
    />
  </div>
</DocumentFragment>
`;
